json.go

```diff
@@ -0,0 +1,100 @@
+package clover
+
+import (
+	"os"
+	"encoding/json"
+	"bufio"
+	"io/ioutil"
+	"errors"
+)
+
+// Exporting errors
+var (
+	ErrImportPathUnreachable = errors.New("the import path can't be reached")
+	ErrExportPathExists = errors.New("the export path already exists")
+	ErrExportCollectionEmpty = errors.New("the collection is empty")
+
+)
+
+func pathExists(path string) bool{
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+// Exports a `collection` to JSON format in `filePath`
+func (db *DB) ExportCollection(collectionName string, exportPath string) error{
+	exists, err := db.HasCollection(collectionName)
+	if err != nil{
+		return err
+	}
+	if !exists{
+		return ErrCollectionNotExist
+	}
+	if pathExists(exportPath) {
+    	return ErrExportPathExists
+	}
+
+	count, err := db.Query(collectionName).Count()
+	if err != nil{
+		return err
+	}
+	if count == 0{
+		return ErrExportCollectionEmpty
+	}
+
+	result, err := db.Query(collectionName).FindAll()
+	if err != nil{
+		return err
+	}
+
+	docs := make([]map[string]interface{}, 0)
+	for _, doc := range result{
+		docs = append(docs, doc.fields)
+	}
+
+	jsonString, _ := json.Marshal(docs)
+	if err != nil{
+		return err
+	}
+
+	ioutil.WriteFile(exportPath, jsonString, os.ModePerm)
+	return nil
+}
+
+// Creates a new `collection` from a JSON file specified by `filePath`
+func (db *DB) ImportCollection(collectionName string, importPath string) error {
+	if !pathExists(importPath){
+        return ErrImportPathUnreachable
+    }
+
+	file, err := os.Open(importPath)
+	if err != nil {
+		return err
+	}
+
+	reader := bufio.NewReader(file)
+	decoder := json.NewDecoder(reader)
+	
+	_, err = decoder.Token()
+	if err != nil{
+		return err
+	}
+
+	docs := make([]*Document, 0)
+	for decoder.More(){
```

Comment:
Since you are loading all data in memory, before calling `Insert()`, deserializing documents one by one only results in a more complicated code. You could simply use the method `Decode()` to deserialize the entire file to a `[]map[string]interface{}` at once which you later transform into a `[]*Document`.